drivers/interrupt_controller/Kconfig.plic

```diff
@@ -13,11 +13,45 @@ config PLIC
 
 if PLIC
 
+config PLIC_IRQ_AFFINITY
+	bool "PLIC IRQ affinity"
+	default y
+	depends on SMP
+	depends on MP_MAX_NUM_CPUS > 1
+	select RISCV_PLIC_HAS_IRQ_AFFINITY
+	help
+	  Enable IRQ affinity.
+
+config PLIC_IRQ_AFFINITY_MASK
+	hex "IRQ affinity mask"
+	depends on PLIC_IRQ_AFFINITY
+	default 0x1
```

Comment:
I kind of wonder whether the default mask should be 0xffffffff, which would (presumably) allow irqs to be handled by the first available core in an SMP configuration.

Another random thought - it would be kind of ideal if the implementation would also unpend the IRQ on other cores after "winning the race" (if that isn't done at the hardware level, and the arch has the capability to do so in hw).

Additionally, why not make this mask option arch-independent / generic?

Reply:
> I kind of wonder whether the default mask should be 0xffffffff, which would (presumably) allow irqs to be handled by the first available core in an SMP configuration.

That's debatable, I left the default as `0x1` as that's the default behavior when IRQ affinity isn't enabled

> Another random thought - it would be kind of ideal if the implementation would also unpend the IRQ on other cores after "winning the race" (if that isn't done at the hardware level, and the arch has the capability to do so in hw).

On RISCV, there seems to be no way for the core that successfully claimed that IRQ to stop the PLIC from waking up other cores.

<img width="646" alt="RISCV Interrupt Notification" src="https://github.com/user-attachments/assets/a0feff7a-9f57-4906-a2b1-a1eaf2577e8f">

> Additionally, why not make this mask option arch-independent / generic?

Makes sense, will do that soon

Reply:
> That's debatable, I left the default as `0x1` as that's the default behavior when IRQ affinity isn't enabled

Is 0x1 the default for all architectures or just RISC-V?

Future thing to consider though. I don't think it's a big deal.

Reply:
> > That's debatable, I left the default as `0x1` as that's the default behavior when IRQ affinity isn't enabled
> 
> Is 0x1 the default for all architectures or just RISC-V?

It will be the default value for all arch when I make the Kconfig generic

Reply:
I guess my question is whether all architectures currently only allow a single core in a cluster to receive interrupts.

Reply:
> I guess my question is whether all architectures currently only allow a single core in a cluster to receive interrupts.

Sorry, no idea about that yet 😬, hopefully other arch maintainers can answer

Reply:
I've moved that Kconfig to `arch/Kconfig` and removed the default value, so each implementation is free to define their own default, or leave it to the user when they set `CONFIG_IRQ_AFFINITY=y`.

The default value for RISC-V is defined to be `0x1` in `Kconfig.plic`